packages/mdc-shape/foundation.js

```diff
@@ -0,0 +1,419 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {MDCFoundation} from '@material/base';
+
+import {cssClasses, strings, numbers} from './constants';
+
+export default class MDCShapeFoundation extends MDCFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      setCanvasWidth: ( /* value: number */ ) => {},
+      setCanvasHeight: ( /* value: number */ ) => {},
+      getCanvasWidth: () => /* number */ 0,
+      getCanvasHeight: () => /* number */ 0,
+      getDevicePixelRatio: () => /* number */ 1,
+      create2dRenderingContext: () =>
```

Comment:
closure is not complaining yet, but it will once we add it to the whitelist, and then it will need all these methods. So I think I'll leave them for that day.